src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,137 @@
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.utility.RegistrationException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static List<User> storage;
+    private static User initialUser;
+    private static User properToRegistUser;
+
+    @BeforeAll
+    static void initializeRegistrationService() {
+        registrationService = new RegistrationServiceImpl();
+        storage = new ArrayList<>();
+        properToRegistUser = new User();
+    }
+
+    @BeforeEach
+    void setUp() {
+        initialUser = new User();
+        properToRegistUser.setLogin("dummy@37");
+        properToRegistUser.setAge(18);
+        properToRegistUser.setPassword("password1234");
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.remove(properToRegistUser);
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        initialUser = null;
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        initialUser.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_loginLessThanSixSymbols_notOk() {
+        initialUser.setLogin("dummy");
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_loginHasSixOrMoreSymbols_ok() {
+        initialUser.setLogin("dummy@37");
+        initialUser.setAge(18);
+        initialUser.setPassword("password1234");
+        User userToCheck = registrationService.register(initialUser);
+        boolean expected = userToCheck.getLogin().length() > 5;
+        assertTrue(expected);
+    }
+
+    @Test
+    void register_loginIsEmptyString_notOk() {
+        initialUser.setLogin("");
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_ageLessThan18Years_notOk() {
+        initialUser.setAge(17);
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_ageIsNegativeNumber_notOk() {
+        initialUser.setAge(-12);
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_ageIsEqualTo18YearsOrMore_ok() {
+        initialUser.setLogin("dummy@37");
+        initialUser.setAge(18);
+        initialUser.setPassword("password1234");
+        User userToCheck = registrationService.register(initialUser);
```

Comment:
You are not asserting against the actual result of the registrationService.register method. Consider adding an assertEquals to compare the expected and actual User object returned by the register method.